### PR TITLE
Landing page donate

### LINKED ISSUE #117: Create the Donate block
- Use the image of the Salvia team
- The button should open a new tab/window and link to paypal

Screen Shot 2020-05-12 at 11 52 47

### PR DESCRIPTION
**Pull request by lehysen**

### CODE REVIEW

**Comment by misscs** [comment]
@lehysen Few things to remember

1. When you're PR is ready for review, request the reviewer so that they are notified
2. Always title case your PR commit messages
3. We use the present/imperative tense for PR commit messages, not the past tense. You want to answer the question _This PR...._ in your minds. Example, Agrega instead of Agregando

File: app/assets/styles/styles.css [hunk 1]
```diff
@@ -170,3 +170,26 @@ a {
...
+  text-align: center;
+}
+
+.donate__btn {
+  font-size: 1.5rem;
+  font-weight: bold;
+  text-decoration: none;
+  color: white;
+  background-color: #6fcf97;
+  padding: 15px 40px;
```
Comment (misscs) [inline app/assets/styles/styles.css]: We define padding in `em` values

File: index.html [hunk 2]
```diff
@@ -87,7 +87,13 @@ <h1 class="logo-brand" itemprop="brand">MACILE</h1>
       </div>
     </div>
   </nav>
-
+  <div class="donate">
+    <div>
```
Comment (misscs) [inline index.html]: Why do we need another `div` inside the parent?

File: app/assets/styles/styles.css [hunk 3]
```diff
@@ -170,3 +170,26 @@ a {
     font-size: 1.25rem;
   }
 }
+
+.donate {
+  height: 300px;
+  font-size: 1.5rem;
+  margin-left: 2em;
+  margin-right: 2em;
+  background-image: linear-gradient(rgba(21, 21, 78, 0.5), rgba(21, 21, 78, 0.6)), url("/home/salvia/news.macile.org/app/assets/images/salvia-team.webp");
```
Comment (misscs) [inline app/assets/styles/styles.css]: The image path needs to be absolute from the project directory not your computer directory

**Review by misscs — CHANGES_REQUESTED** [review]
A few small changes. 

Can you push your work to surge? You will need to install it on your machine following these directions here ... http://surge.sh/help/getting-started-with-surge Step #2. Your computer already has node installed. Just follow the command. 

Once you install surge you will need to create an account. Then you can publish your pages.

File: app/assets/styles/styles.css (hunk 1 above)
Comment (lehysen) [reply app/assets/styles/styles.css]: **Domain:** hesitant-ink.surge.sh

Ese es el dominio en surge.

File: app/.vscode/settings.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+    "git.ignoreLimitWarning": true
+}
```
Comment (misscs) [inline app/.vscode/settings.json]: This file should not be in the repo. Make sure you don't use `git add .` I always like to double check what files changed. 

You may need to add this file to `.gitignore`. This is a file that that lives in the project root.

File: package.json [hunk 5]
```diff
@@ -24,5 +24,8 @@
     "imagemin-mozjpeg": "^8.0.0",
     "imagemin-pngquant": "^8.0.0",
     "imagemin-webp": "^5.1.0"
+  },
+  "dependencies": {
+    "http-server": "^0.12.3"
```
Comment (misscs) [inline package.json]: We should be careful not to add `http-server` as a project dependency. That means that it is required by the project in production. 

In general, I don't think we want to add it as a dev dependency either. It is just a tool that you need on your machine. 

You need to remove lines 28-30. Then run `npm install` so it removes the dependency from `package-lock.json`. Commit the changes to `package-lock.json`. 

You probably need to install the server globally `npm install http-sever -g`. That makes it global on your machine.

**Comment by misscs** [comment]
Don't forget that your commit messages need to start with a capital letter!

**Comment by misscs** [comment]
Thank you for pushing your work to surge. It looks pretty good. I would like to request a few changes so that that design really pops. 

## Add more space between the text and button

In the mockup, the Donate button is not flush with the text.
- Add `display: block` to the donate button. This ensures that the link behaves like a block level element. Without this, if you add margin to the `p` it won't visually apply.
- Add a class to the `p` element and then you can add margin bottom

## Don't rely on the browser defaults

Your code is relying on the browser default styling for `h2` and `p` tag. The browser defaults look okay on large screen devices but it is too big on mobile. 

- Add a class to each of these elements following BEM conventions
- Then you can reset these elements. You want to consider adding font-size and line-height to each of them.

**Review by misscs — CHANGES_REQUESTED** [review]

File: app/assets/styles/styles.css [hunk 6]
```diff
@@ -170,3 +170,26 @@ a {
     font-size: 1.25rem;
   }
 }
+
+.donate {
+  height: 300px;
```
Comment (misscs) [inline app/assets/styles/styles.css]: In general, it is best to avoid a fixed height on elements. I like to use padding because it creates a more fluid scaling experience. 300px may look good on desktop, but it takes up too much space on mobile. 

Let's remove this property and add `padding: 1em 3em; for large screens and `padding: 1em` for small screens.

File: app/assets/styles/styles.css [hunk 7]
```diff
@@ -170,3 +170,26 @@ a {
...
+  font-size: 1.5rem;
```
Comment (misscs) [inline app/assets/styles/styles.css]: It isn't recommended to apply a font-size to the parent element unless all the children use the same font-size or we are using ems/% based fonts and want to scale from a parent/root element. 

Since our code base is more explicit, we want to apply `font-size` directly on the elements we want to target, the title, subtitle, and button, in the case of the donate box.

File: app/assets/styles/styles.css [hunk 8]
```diff
@@ -170,3 +170,26 @@ a {
...
+  margin-left: 2em;
+  margin-right: 2em;
```
Comment (misscs) [inline app/assets/styles/styles.css]: We're going to need to define a max-width. I didn't really think about one when I making the designs. For now, because we know it will need to sit in the center let's replace `2em` with `auto`. 

Once we merge everything in, I'll take a cleanup pass to add max-widths to containers.

File: app/assets/styles/styles.css [hunk 9]
```diff
@@ -170,3 +170,26 @@ a {
...
+  background-size: cover;
+  background-position: center center;
+  background-repeat: no-repeat;
+  border: 5px solid #096a97;
+  color: #fff;
+  text-align: center;
+}
+
+.donate__btn {
+  font-size: 1.5rem;
```
Comment (misscs) [inline app/assets/styles/styles.css]: This font size is a bit big. Let's remove it for now.

File: app/assets/styles/styles.css [hunk 10]
```diff
@@ -170,3 +170,26 @@ a {
...
+  font-weight: bold;
+  text-decoration: none;
+  color: white;
+  background-color: #6fcf97;
+  padding: 1em 1.5em;
```
Comment (misscs) [inline app/assets/styles/styles.css]: There's a bit too much padding on the button. Try `0.5em 1em`

File: app/assets/styles/styles.css [hunk 11]
```diff
@@ -170,3 +170,26 @@ a {
...
+  background-repeat: no-repeat;
+  border: 5px solid #096a97;
+  color: #fff;
+  text-align: center;
+}
+
+.donate__btn {
+  font-size: 1.5rem;
+  font-weight: bold;
+  text-decoration: none;
```
Comment (misscs) [inline app/assets/styles/styles.css]: I believe this is applied at the global level. Let's remove it for now. If a line appears, I am confident it will go away once we merge everything together :)

File: index.html [hunk 12]
```diff
@@ -87,7 +87,11 @@ <h1 class="logo-brand" itemprop="brand">MACILE</h1>
       </div>
     </div>
   </nav>
-
+  <div class="donate">
+      <h2>Support our growth!</h2>
```
Comment (misscs) [inline index.html]: Let's use the custom font for the `h2`. The rest of the text will inherit `Roboto`.

**Review by misscs — CHANGES_REQUESTED** [review]

File: app/assets/styles/styles.css [hunk 13]
```diff
@@ -172,24 +172,36 @@ a {
...
+  margin-right: auto;
+}
+.donate__paragraph {
+  width:25em;
+  font-size: 1.5rem;
+  margin-bottom: 1.5em;
+  margin-left: auto;
+  margin-right: auto;
+}
+.donate__h2 {
```
Comment (misscs) [inline app/assets/styles/styles.css]: Getting there! 

1. We always have a space between each ruleset in our code base. 
1. We prefer to use semantic class names rather than use the html element. 

There are many examples of how we are naming classes in the code base. Take a look at the bulletin code or some of the PRs #127

File: app/assets/styles/styles.css [hunk 14]
```diff
@@ -172,24 +172,36 @@ a {
...
-  padding: 1em 1.5em;
+  padding: 0.5em 1em;
+  display: block;
+  width:3em;
+  font-size: 1.5rem;
+  margin-left: auto;
+  margin-right: auto;
+}
+.donate__paragraph {
+  width:25em;
```
Comment (misscs) [inline app/assets/styles/styles.css]: Always a space after the semicolon

File: package.json [hunk 15]
```diff
@@ -24,8 +24,4 @@
     "imagemin-mozjpeg": "^8.0.0",
     "imagemin-pngquant": "^8.0.0",
     "imagemin-webp": "^5.1.0"
-  },
-  "dependencies": {
-    "http-server": "^0.12.3"
-  }
```
Comment (misscs) [inline package.json]: You need to run `npm install` to remove the dependency from the `package-lock.json` file. Otherwise it is still technically a dependency of the project.

File: app/assets/styles/styles.css [hunk 16]
```diff
@@ -172,24 +172,36 @@ a {
...
 .donate__btn {
-  font-size: 1.5rem;
-  font-weight: bold;
-  text-decoration: none;
   color: white;
   background-color: #6fcf97;
-  padding: 1em 1.5em;
+  padding: 0.5em 1em;
+  display: block;
+  width:3em;
```
Comment (misscs) [inline app/assets/styles/styles.css]: If you use `inline-block` you won't need to define a width. Try it out.

**Review by misscs — CHANGES_REQUESTED** [review]
Getting there! ⭐  Definitely improved, but still work to be done. Take you time. I left many of the same comments on the first review. One suggestion is to write down a list of the changes in the description then you can be sure you capture everything. The github comments are hard to follow sometimes.

File: app/assets/styles/styles.css (hunk 13 above)
Comment (misscs) [reply app/assets/styles/styles.css]: Resolved pairing

**Review by misscs — APPROVED** [review]
